src/analysis/calculus/times_cont_diff.lean

```diff
@@ -2714,6 +2714,45 @@ lemma times_cont_diff.has_strict_deriv_at
   has_strict_deriv_at f (deriv f x) x :=
 hf.times_cont_diff_at.has_strict_deriv_at hn
 
+lemma has_ftaylor_series_up_to_on.exists_lipschitz_on_with {E F : Type*} [normed_group E]
+  [normed_space ℝ E] [normed_group F] [normed_space ℝ F] {f : E → F}
+  {p : E → formal_multilinear_series ℝ E F} {s : set E} {x : E}
+  (hf : has_ftaylor_series_up_to_on 1 f p (insert x s)) (hs : convex s) (K : ℝ≥0)
+  (hK : ∥p x 1∥₊ < K) :
+  ∃ t ∈ 𝓝[s] x, lipschitz_on_with K f t :=
+begin
+  set f' := λ y, continuous_multilinear_curry_fin1 ℝ E F (p y 1),
+  have hder : ∀ y ∈ s, has_fderiv_within_at f (f' y) s y,
+    from λ y hy, (hf.has_fderiv_within_at le_rfl (subset_insert x s hy)).mono (subset_insert x s),
+  have hcont : continuous_within_at f' s x,
+    from (continuous_multilinear_curry_fin1 ℝ E F).continuous_at.comp_continuous_within_at
+      ((hf.cont _ le_rfl _ (mem_insert _ _)).mono (subset_insert x s)),
+  replace hK : ∥f' x∥₊ < K, by simpa only [linear_isometry_equiv.nnnorm_map],
+  exact hs.exists_nhds_within_lipschitz_on_with_of_has_fderiv_within_at_of_continuous_within_at
+    (eventually_nhds_within_iff.2 $ eventually_of_forall hder) hcont K hK
+end
+
+lemma times_cont_diff_within_at.exists_lipschitz_on_with {E F : Type*} [normed_group E]
+  [normed_space ℝ E] [normed_group F] [normed_space ℝ F] {f : E → F} {s : set E}
+  {x : E} (hf : times_cont_diff_within_at ℝ 1 f s x) (hs : convex s) :
+  ∃ (K : ℝ≥0) (t ∈ 𝓝[s] x), lipschitz_on_with K f t :=
+begin
+  rcases hf 1 le_rfl with ⟨t, hst, p, hp⟩,
+  rcases no_top (∥p x 1∥₊) with ⟨K, hK⟩,
+  rcases metric.mem_nhds_within_iff.mp hst with ⟨ε, ε0, hε⟩,
+  replace hp : has_ftaylor_series_up_to_on 1 f p (metric.ball x ε ∩ insert x s) := hp.mono hε,
+  clear hst hε t,
+  rw [← insert_eq_of_mem (metric.mem_ball_self ε0), ← insert_inter] at hp,
+  rcases hp.exists_lipschitz_on_with ((convex_ball _ _).inter hs) K hK with ⟨t, hst, hft⟩,
+  rw [inter_comm, ← nhds_within_restrict' _ (metric.ball_mem_nhds _ ε0)] at hst,
+  exact ⟨K, t, hst, hft⟩
+end
+
+lemma times_cont_diff_at.exists_lipschitz_on_with {f : E' → F'} {x : E'}
```

Comment:
There is an API difference between `times_cont_diff_within_at.exists_lipschitz_on_with` (where the conclusion is that there exists some `K` such that ...) and this one, where `K` is part of the assumptions, in terms of `fderiv`. Can you have the same form for both of them? (Or 4 lemmas, giving the two forms for each of them)?

Reply:
In the previous lemma I can't have a version without `∃ K` because the lower bound comes from unpacking `∃` in `times_cont_diff_within_at`. I can add versions with `∃ K` to other lemmas, if you want.

Reply:
Yes, adding the `∃ K` version to the other lemma would be perfect.